pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -783,3 +783,98 @@ func (s *Session) UpdateVirtualMachine(
 	// TODO: Find a better place for this?
 	return s.attachClusterModule(vmCtx, resVM, updateArgs.ResourcePolicy)
```

Comment:
`updateArgs.ResourcePolicy` is only set when L759 is hit so for like a TKG VM that has the cluster module annotation, we'd NPE when trying to find/attach the CM.

I understand that we're to do something quick here, but ay scope that we now have a partially initialized `updateArgs` would make me worried we have an NPE either now or later due to how messy & tangled the existing code is.